presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -359,6 +362,49 @@ public static Slice splitPart(@SqlType(StandardTypes.VARCHAR) Slice string, @Sql
         return null;
     }
 
+    @Description("creates a map using delimiter1 and delimiter2")
+    @ScalarFunction
+    @SqlType("map<varchar,varchar>")
+    public static Slice splitToMap(@SqlType(StandardTypes.VARCHAR) Slice string, @SqlType(StandardTypes.VARCHAR) Slice delimiter1, @SqlType(StandardTypes.VARCHAR) Slice delimiter2)
+    {
+        checkCondition(delimiter1.length() > 0, INVALID_FUNCTION_ARGUMENT, "The delimiter1 may not be the empty string");
+        checkCondition(delimiter2.length() > 0, INVALID_FUNCTION_ARGUMENT, "The delimiter2 may not be the empty string");
+
+        Map<Slice, Slice> map = new HashMap<>();
```

Comment:
Don't use `HashMap` as intermediate representation. Construct a `BlockBuilder` using `VARCHAR.createBlockBuilder`. And then use `VARCHAR.writeSlice` to append key1, value1, key2, value2, etc into the Block. At the end, return the result you got from `.build()`. By the way, the return type of functions that return SQL map is changed to `Block` as a result of recent major refactor.